test/aleph/util_test.clj

```diff
@@ -0,0 +1,45 @@
+(ns aleph.util-test
+  (:require [aleph.util :as util]
+            [clojure.test :refer [deftest is testing]]
+            [manifold.deferred :as d]))
+
+(deftest test-propagate-error
+  (testing "Happy path"
+    (let [src (d/deferred)
+          dst (d/deferred)
+          prp (promise)]
+      (util/propagate-error src dst (fn [e] (deliver prp e)))
+      (d/error! src :boom)
+      (is (d/realized? dst))
```

Comment:
Isn't this a race condition? Might need to put them on same executor and thread.